lib/llist.c

```diff
@@ -191,9 +192,11 @@ Curl_node_uremove(struct Curl_llist_node *e, void *user)
 
   list = e->_list;
   DEBUGASSERT(list);
-  ptr = Curl_node_take_elem(e);
-  if(list->_dtor)
-    list->_dtor(user, ptr);
+  if(list) {
```

Comment:
If take_elem checks, this is not necessary, or?

Reply:
It needs it since it references the pointer in `list->_dtor`.